jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1259,10 +1259,12 @@ protected boolean parseFields(ByteBuffer buffer)
             if (_maxHeaderBytes > 0 && ++_headerBytes > _maxHeaderBytes)
             {
                 boolean header = _state == State.HEADER;
-                LOG.warn("{} is too large {}>{}", header ? "Header" : "Trailer", _headerBytes, _maxHeaderBytes);
-                throw new BadMessageException(header
-                    ? HttpStatus.REQUEST_HEADER_FIELDS_TOO_LARGE_431
-                    : HttpStatus.PAYLOAD_TOO_LARGE_413);
+                if (debugEnabled)
+                    LOG.debug("{} is too large {}>{}", header ? "Header" : "Trailer", _headerBytes, _maxHeaderBytes);
+                if (_requestParser)
+                    throw new BadMessageException(header ? HttpStatus.REQUEST_HEADER_FIELDS_TOO_LARGE_431 : HttpStatus.PAYLOAD_TOO_LARGE_413);
+                // There is no equivalent of 431 for response headers.
+                throw new BadMessageException(HttpStatus.BAD_REQUEST_400, "Response Header Fields Too Large");
```

Comment:
It's not a 400 error either as 4xx codes indicate client errors but here it really is a server error. 500 looks like a more correct response code IMHO.

Reply:
Unfortunately `BadMessageException` only allows `4xx` error codes.

I'll check if we can throw a different exception.

Reply:
It is possible to throw a different exception, `HttpException.RuntimeException`, but this is a client parsing a response: should it really generate a 500 Internal _Server_ Error (emphasis on "Server")?

The server could generate a large response; it's the client that cannot parse it, so it's not technically a server error in HTTP/1.1 where the client cannot inform the server about how much it can parse.

@gregw thoughts?